### PR TITLE
[JENKINS-65127] Support IMDSv2 configuration

### PR DESCRIPTION
**Pull request by warrior8792**
Add metadata configuration options in the Advanced section

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by warrior8792** [comment]
New fields added in the advanced section

Screenshot_2021-05-04 Configure Clouds  Jenkins

File: src/main/java/hudson/plugins/ec2/SlaveTemplate.java [hunk 1]
```diff
@@ -224,6 +230,12 @@
 
     public EbsEncryptRootVolume ebsEncryptRootVolume;
 
+    public Boolean metadataEndpointEnabled;
```
[the remark this answers is not in the document]
Comment (warrior8792) [reply src/main/java/hudson/plugins/ec2/SlaveTemplate.java]: Made these private and added getters

**Comment by avivek** [comment]
Looking forward to this feature.

**Comment by avivek** [comment]
Also another question is the EC2 plugin's behaviour on the master.
The master is attached with instance profile and in such a case will this plugin use the IMDSv2 instead of V1 to obtain credentials and such.
The AWS java sdk would use IMDSv2 if available yet something to look out for.

File: src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java [hunk 2]
```diff
@@ -104,6 +108,10 @@
     public final Tenancy tenancy;
     private String instanceType;
 
+    public final Boolean metadataEndpointEnabled;
```
Comment (res0nance) [inline src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java:111]: can you make these field private?

File: src/main/resources/hudson/plugins/ec2/SlaveTemplate/config.jelly [hunk 3]
```diff
@@ -223,6 +223,15 @@ THE SOFTWARE.
 
     <f:descriptorList title="${%Node Properties}" field="nodeProperties" descriptors="${descriptor.nodePropertyDescriptors}" />
 
+    <f:entry title="${%Enable Metadata HTTP Endpoint}" field="metadataEndpointEnabled" >
+      <f:checkbox default="true"/>
+    </f:entry>
+    <f:entry title="${%Metadata Require HTTP Tokens}" field="metadataTokensRequired">
+      <f:checkbox default="false"/>
+    </f:entry>
+    <f:entry title="${%Metadata Put Response Hop Limit}" field="metadataHopsLimit">
```
Comment (res0nance) [inline src/main/resources/hudson/plugins/ec2/SlaveTemplate/config.jelly:232]: Are there any restrictions to the value in this field?

File: src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java (hunk 2 above)
Comment (warrior8792) [reply src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java]: 739b04fe

File: src/main/resources/hudson/plugins/ec2/SlaveTemplate/config.jelly (hunk 3 above)
Comment (warrior8792) [reply src/main/resources/hudson/plugins/ec2/SlaveTemplate/config.jelly:232]: 1-64 per Amazon.  ae7ba08

**Review by res0nance — APPROVED** [review]

**Pull request merged**